src/vs/editor/contrib/find/common/findController.ts

```diff
@@ -378,7 +391,21 @@ export class NextMatchFindAction extends MatchFindAction {
 	}
 
 	protected _run(controller: CommonFindController): boolean {
```

Comment:
Here we are always using the search string found in the global search clipboard, but we write to the global search clipboard only in the `StartSearchAction`. Perhaps the controller should write to the global search clipboard on each and every `findState` change ? i.e. anytime the user types in the find input ?

Reply:
Thanks a lot for the review @alexandrudima . This is a great suggestion to set the global buffer as and when the find input changes. 

I see that the when the find string changes in the find input and enter is hit, the search resets. I will look into this first.

Reply:
Everytime time the user types in the find input, I am writing to the global buffer . Also, everytime the user changes the global buffer using some other app and returns back to vscode, if the find box is open the find search string changes based on the global buffer